src/umpire/ResourceManager.cpp

```diff
@@ -154,7 +162,13 @@ ResourceManager::initialize()
     resource::MemoryResourceRegistry::getInstance();
 
   {
-    std::unique_ptr<strategy::AllocationStrategy> host_allocator{registry.makeMemoryResource("HOST", getNextId())};
+    std::unique_ptr<strategy::AllocationStrategy> 
+      host_allocator{
+        util::wrap_allocator<
+          strategy::AllocationTracker,
+          strategy::ZeroByteHandler>(
+            registry.makeMemoryResource("HOST", getNextId()))};
```

Comment:
If you were to pass the internal umpire pool here to the constructor, then the circular dependency is broken.